### ISSUE TITLE
Widget is not updated when task become "current tasks"

### ISSUE THREAD

**Issue by dmfs** [dmfs/opentasks#17]
The widget should be updated if the start date or the due date of a task has passed.

The widget should determine the next upcoming start date or due date and set an alarm.

**Comment by wweich** [comment]
I don't know if I should create a new issue or reuse this one, but I have this exact problem on my Nexus 5 and Nexus 7 (Android 5.1, stock).
I get notifications for start and due date but the widget only updates on data change (new/edit/delete) but not when the start date or due date of a task passes.
It's the same on the default Google Now Launcher and Action Launcher 3.
Killing the Launcher will (of cause) also update the widget.

**Comment by dmfs** [comment]
@wweich we'll look into that again.

**Comment by Trogel** [comment]
I can still reproduce this issue, im my case on LineageOS 16.0-20200309 (Android 9) using a Honor 5X.

In addition, a task in the widget does not change its color to red when it becomes due, unless an update is triggered e.g. due to synchronization or by manually scrolling down and up through the list of tasks. At the same time, notifications work as expected.

I looked into the code and noticed that the `TaskListWidgetUpdaterService` only listens for `ACTION_TIMEZONE_CHANGED` and `ACTION_TIME_CHANGED`, whereas the notifications rely on `ACTION_BROADCAST_TASK_DUE` and `ACTION_BROADCAST_TASK_STARTING`. (According to [an answer on StackOverflow](https://stackoverflow.com/questions/7912885/action-time-changed-or-action-date-changed-cant-make-them-work/19112483#19112483) `ACTION_TIMEZONE_CHANGED` and `ACTION_TIME_CHANGED` "are triggered only when the user manually changes time or date, and they are not automatically triggered by the system".)

Therefore I changed `ActionService` to also update all widgets when receiving `ACTION_BROADCAST_TASK_DUE` or `ACTION_BROADCAST_TASK_STARTING`. For me this change resolves the issue. I'll provide a pull request soon.
